src/overlay/Peer.cpp

```diff
@@ -822,22 +826,38 @@ Peer::recvMessage(StellarMessage const& stellarMsg)
     // group messages used during handshake, process those synchronously
     case HELLO:
     case AUTH:
+        // case ERROR_MSG:
         Peer::recvRawMessage(stellarMsg);
         return;
     case SEND_MORE:
+    case SEND_MORE_EXTENDED:
     {
         cat = "CTRL";
 
-        if (stellarMsg.sendMoreMessage().numMessages == 0)
+        if (getNumMessages(stellarMsg) == 0 ||
+            (mFlowControlBytes &&
+             stellarMsg.sendMoreExtendedMessage().numBytes == 0))
         {
             drop("unexpected SEND_MORE message",
                  Peer::DropDirection::WE_DROPPED_REMOTE,
                  Peer::DropMode::IGNORE_WRITE_QUEUE);
             return;
         }
 
-        if (stellarMsg.sendMoreMessage().numMessages >
-            UINT64_MAX - mOutboundCapacity)
+        auto overflow =
+            mFlowControlMessages->getOutboundCapacity() > 0 &&
```

Comment:
We don't, message outbound capacity is guaranteed to always be positive (I initially made this change when I made outbound capacity int64 instead of uint64, but forgot to remove it). Removed it.